File: Trackly.APP/trackly_app/lib/src/data/models/work_item.dart
class WorkItem {
  String title;
  String description;
  String priority;
  String category;
  String creatorUserId;
  String assetId;

  WorkItem({
    required this.title,
    required this.description,
    required this.priority,
    required this.category,
    required this.creatorUserId,
    required this.assetId,
  });

  //Convert the work item to a json object
  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'priority': priority,
      'category': category,
      'creatorUserId': creatorUserId,
      'assetId': assetId,
    };
  }

  //Create a work item for from a json object
  factory WorkItem.fromJson(Map<String, dynamic> json) {
    return WorkItem(
      title: json['title'],
      description: json['description'],
      priority: json['priority'],
      category: json['category'],
      creatorUserId: json['creatorUserId'],
      assetId: json['assetId'],
    );
  }
}

File: Trackly.API/Models/Tickets/WorkItem.cs
using Newtonsoft.Json;
using TracklyApi.Models.Assets;
using static TracklyApi.Helpers.EnumHelper;

namespace TracklyApi.Models.Tickets
{
    public class WorkItem
    {
        public Guid WorkItemId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public WorkItemStatus Status { get; set; }
        public Priority Priority { get; set; }
        public TicketCategory Category { get; set; }
        public Guid? CreatorUserID { get; set; }
        public Guid AssetId { get; set; }

        //Time created for the ticket
        public DateTime CreatedAt { get; set; }

        // Navigation properties
        [JsonIgnore]    
        public virtual Asset Asset { get; set; }
    }
}
